### PR TITLE
OSDOCS-2400: Document how to create custom domains for applications.

### PR DESCRIPTION
**Pull request by jonquilwilliams**
JIRA: https://issues.redhat.com/browse/OSDOCS-2400

**Previews:** 

- **ROSA:** https://deploy-preview-36131--osdocs.netlify.app/openshift-rosa/latest/applications/deployments/osd-config-custom-domains-applications.html

- **OSD:** https://deploy-preview-36131--osdocs.netlify.app/openshift-dedicated/latest/applications/deployments/osd-config-custom-domains-applications.html

Branch = dedicated-4. No CP and no milestones.

### CODE REVIEW

**Comment by 0kashi** [comment]
second bullet in the prereqs has a superfluous " ` " in the beginning before "*.apps".  Otherwise lgtm.

**Comment by yuwang-RH** [comment]
LGTM.

File: applications/deployments/osd-config-custom-domains-applications.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[id="osd-config-custom-domains-applications"]
+= Configuring custom domains for applications
+include::modules/attributes-openshift-dedicated.adoc[]
+:context: osd-config-custom-domains-applications
+
+toc::[]
+
+include::modules/osd-applications-config-custom-domains.adoc[leveloffset=+2]
```
Comment (maxwelldb) [inline applications/deployments/osd-config-custom-domains-applications.adoc:8]: The leveloffset=+2 makes this render as an H3 in your preview. Is that intentional?

File: modules/osd-applications-config-custom-domains.adoc [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+Do not use the reserved names `default` or `apps*`, such as `apps` or `apps2`, in the `metadata/name:` section of the `CustomDomain` CR.
+
+.Procedure
+
+. Create a new TLS secret from a private key and a public certificate, where `fullchain.pem` and `privkey.pem` are your public or private wildcard certificates.
++
+.Example
++
+
+[source,terminal]
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:27]: Two thoughts!

1. AsciiDoc titles will apply to code blocks without the `+`. You should be able to eliminate lines 25 and 26 safely. Same idea throughout the PR. 
2. Do you just want the reader to run the command you have here? If so, you could tweak line 22 a bit, end it with a colon, and eliminate lines 24-26.

File: modules/osd-applications-config-custom-domains.adoc [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+----
+$ oc create secret tls <name>-tls --cert=fullchain.pem --key=privkey.pem -n <my_project>
+----
+
+. Create a new `CustomDomain` custom resource (CR). For example, using the file `<company_name>-custom-domain.yaml`, `<name>-tls` is the secret created in the previous step, and `apps.companyname.io` is the custom domain.
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:32]: Could this be chopped down by moving the example YAML title into the example title?

You could also use callouts in the example to eliminate the second sentence.

File: modules/osd-applications-config-custom-domains.adoc [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  name: <company_name>
+spec:
+  domain: apps.companyname.io
+  scope: External
+  certificate:
+    name: <name>-tls
+    namespace: <my_project>
+----
+
+. Apply the CR.
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:50]:
```suggestion
. Apply the CR:
```

File: modules/osd-applications-config-custom-domains.adoc [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
++
+.Example
++
+[source,terminal]
+----
+$ oc apply -f <company_name>-custom-domain.yaml
+----
+
+. Get the status of your newly created CR.
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:59]:
```suggestion
. Get the status of your newly created CR:
```

Comment (maxwelldb) [reply modules/osd-applications-config-custom-domains.adoc:59]: Same suggestion across the board.

File: modules/osd-applications-config-custom-domains.adoc [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+----
+NAME               ENDPOINT                                                    DOMAIN                       STATUS
+<company_name>     xxrywp.<company_name>.cluster-01.opln.s1.openshiftapps.com  *.apps.<company_name>.io     Ready
+----
+
+. Using the endpoint value, add a new wildcard CNAME recordset to your managed DNS provider, such as Route53, Azure DNS, or Google DNS.
++
+.Example
++
+[source,terminal]
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:78]: Is this a command? Doesn't look like `[source,terminal]` applies here.

File: modules/osd-applications-config-custom-domains.adoc [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+----
+*.apps.<company_name>.io -> xxrywp.<company_name>.cluster-01.opln.s1.openshiftapps.com
+----
+
+. Create a new application and expose it.
++
+.Example
++
+[source,terminal]
+----
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:88]: This should be broken up into multiple code blocks per [our guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output).

**Review by maxwelldb — COMMENTED** [review]
Made a few comments and suggestions. TY!

File: modules/osd-applications-config-custom-domains.adoc [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+$ oc create route edge --service=hello-openshift hello-openshift-tls --hostname hello-openshift-tls-my-project.apps.acme.io -n my-project
+
+$ oc get route -n my-project
+
+$ curl https://hello-openshift-tls-my-project.apps.<company_name>.io
+Hello OpenShift!
+----
++
+[NOTE]
```
Comment (maxwelldb) [inline modules/osd-applications-config-custom-domains.adoc:99]: Oh, and should this note apply to this step, or to the entire module?

**Pull request merged**